src/components/metadata/index.js

```diff
@@ -7,22 +7,39 @@ import CopyIcon from '../icons/copy';
 import CloseIcon from '../icons/close';
 import MetaDataRow from './metadata-row';
 import MetaDataValue from './metadata-value';
+import MetaDataCode from './metadata-code';
+import MetaCodeToggle from './metadata-code-toggle';
 import {
   getVisibleMetaSidebar,
   getClickedNodeMetaData
 } from '../../selectors/metadata';
 import { toggleNodeClicked } from '../../actions/nodes';
+import { toggleCode } from '../../actions';
 import './styles/metadata.css';
 
 /**
  * Shows node meta data
  */
-const MetaData = ({ visible = true, metadata, onToggleNodeSelected }) => {
+const MetaData = ({
+  visible = true,
+  metadata,
+  codeFlag,
+  visibleCode,
+  onToggleCode,
+  onToggleNodeSelected
+}) => {
   const [showCopied, setShowCopied] = useState(false);
 
+  // Hide code panel when selected metadata changes
+  useEffect(() => onToggleCode(false), [metadata, onToggleCode]);
+
   const isTaskNode = metadata?.node.type === 'task';
-  const isParametersNode = metadata?.node.type === 'parameters';
   const isDataNode = metadata?.node.type === 'data';
+  const isParametersNode = metadata?.node.type === 'parameters';
+
+  const hasCode = Boolean(metadata?.code);
+  const showCodePanel = codeFlag && visible && visibleCode && hasCode;
```

Comment:
same as above - what does `visible` check for in this case, given that it is assigned `true` anyways?

Reply:
The `visible` prop controls if the panel is shown, which is decided by something outside of the component.